src/fsharp/NameResolution.fs

```diff
@@ -3748,11 +3749,27 @@ let rec ResolvePartialLongIdentInModuleOrNamespace (ncenv: NameResolver) nenv is
              mty.TypeDefinitions
              |> List.filter (fun tcref -> not (tcref.LogicalName.Contains(",")))
              |> List.filter (fun tycon -> not (IsTyconUnseen ad g ncenv.amap m (modref.NestedTyconRef tycon)))
-
-         let ilTyconNames = 
-             mty.TypesByAccessNames.Values
-             |> List.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
-             |> Set.ofList
+         
+         let accessibleSubModules =
+             let moduleOrNamespaces =
+                 mty.ModulesAndNamespacesByDemangledName
+                 |> NameMap.range
+
+             if isNil moduleOrNamespaces then [] else
+
+             let ilTyconNames = 
+                 mty.TypesByAccessNames.Values
+                 |> List.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
```

Comment:
Based on what I saw here: https://github.com/Microsoft/visualfsharp/pull/5632/files#diff-dbb9e59b8ac1e1e04e1b00282a91ec97R3587

Do we actually want to use `List.choose` to return a new list only to pass the list into the hashset? Because that's not what was happening in the other PR. Instead we just iterated over the list and added to the hashset if the condition was true.

Reply:
good point